webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/VaultPath.scala

```diff
@@ -0,0 +1,76 @@
+package com.scalableminds.webknossos.datastore.datavault
+
+import com.scalableminds.util.io.ZipIO
+import com.scalableminds.webknossos.datastore.storage.FileSystemCredential
+import net.liftweb.util.Helpers.tryo
+
+import java.net.URI
+import java.nio.file.{FileSystem, LinkOption, Path, Paths, WatchEvent, WatchKey, WatchService}
+
+/*
+VaultPath implements Path so that a drop in replacement is possible while continuing to use Paths for local storage.
+This class does not implement all relevant methods and it might be a good idea to remove the inheritance on Path in the
+future.
+ */
+
+class VaultPath(uri: URI, dataVault: DataVault, fileSystemCredentialOpt: Option[FileSystemCredential]) extends Path {
+
+  def get(key: String, range: Option[Range] = None): Array[Byte] = dataVault.get(key, this, range)
```

Comment:
I’m a bit confused about the `key` – do we need this distinction between the Path’s uri and the key? I would have assumed that the path _is_ the key.
then path.get(key) would instead be (path / key).get()
Are there problems with that?

Reply:
Also, I’d suggest naming the function `readBytes` sinde `get` is very overloaded in the scala world already. (I assume that you took the name from the GET concept in http, which does make sense, but I think it’s a bit confusing here)